### PR TITLE
[DDO-2190] Strip Host header in CromIAM's forwarding

### PR DESCRIPTION
**Pull request by jack-r-warren**
[The ticket](https://broadworkbench.atlassian.net/browse/DDO-2190) has a ton more info and I talked with @aednichols about this--the short version is:
- Whatever is doing the HTTP request to Cromwell (Akka?) does not set the Host header ([per MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host)) if it has been customized beforehand
- We've accidentally been doing that by copying the _request's_ headers that came into CromIAM, which obviously include the Host header correlating to CromIAM itself
- Thus CromIAM sends requests to Cromwell with an incorrect Host header
- This hasn't mattered before because Cromwell's Layer 4 load balancer doesn't care and Cromwell's Apache proxy didn't actually need to do host-based routing because it just forwards everything to one app, Cromwell
- This suddenly matters a lot now because BEEs use an Nginx controller for ingress instead of a GCP Layer 4 load balancer, and _it_ needs to use host-based routing
- TL;DR: CromIAM is proxying to Cromwell wrong-ish and it very much does not work in BEEs

Solution: strip out the Host header just like CromIAM already does for Timeout-Access, and everything is happy. The impact to live environments should be zero because they clearly didn't care about the header before. If this fails anywhere, Argo sees the failure immediately like it did for BEEs, and it sees it in a way that any deployment or promotion would be halted because CromIAM would fail to come online from Argo's perspective.

### CODE REVIEW

**Review by breilly2 — APPROVED** [review]
👍 Makes sense to me. Thanks for doing this!

**Review by cjllanwarne — APPROVED** [review]
Seems to correctly implement "remove host header". I didn't quite follow the chain of "x causes y causes a problem", but happy to take your word for it 😄
